adlfs/spec.py

```diff
@@ -2168,7 +2167,7 @@ async def _async_upload_chunk(self, final: bool = False, **kwargs):
                 # which is throws an InvalidHeader error from Azure, so instead
                 # of staging a block, we directly upload the empty blob
                 # This isn't actually tested, since Azureite behaves differently.
-                if block_id == "0000000" and length == 0 and final:
+                if block_id == self._get_block_id([]) and length == 0 and final:
```

Comment:
Hi, I've changed the implementation.

Now, if `block_list` is empty, the `block_id` is `"0" * 32`; otherwise, `uuid4().hex`. Both are strings of [length 32](https://docs.python.org/3/library/uuid.html#uuid.UUID.hex).

Also, empty block lists always map to the same result `"0" * 32`, which should keep this line work.

Reply:
That sounds like the right solution to me.